tests/test_iam.py

```diff
@@ -766,6 +766,120 @@ def test_iam_user_access_key_multi(self):
         self.assertEqual(
             resources[0]['c7n:matched-keys'][0]['c7n:match-type'], 'access')
 
+    def test_iam_user_ssh_key_filter(self):
+        factory = self.replay_flight_data('test_iam_user_ssh_key_filter')
+        p = self.load_policy({
+            'name': 'iam-user-ssh-keys',
+            'resource': 'iam-user',
+            'filters': [
+                {'type': 'ssh-key', 'key': 'Status', 'value': 'Active'},
+            ]},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 2)
+        self.assertEqual(len(resources[0]['c7n:matched-ssh-keys']), 1)
+
+    def test_iam_user_ssh_key_age_filter(self):
+        factory = self.replay_flight_data('test_iam_user_ssh_key_age_filter')
+        p = self.load_policy({
+            'name': 'iam-user-old-ssh-keys',
+            'resource': 'iam-user',
+            'filters': [
+                {'type': 'ssh-key', 'key': 'Status', 'value': 'Active'},
+                {'type': 'ssh-key', 'key': 'UploadDate',
+                 'value_type': 'age', 'value': 90, 'op': 'greater-than'},
+            ]},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
rather then refetching the exact same data multiple times via recorded output, ie. recorded json bloat it might be useful to search for the exact resource using a config source, or if its client side filtering, on reuse of recorded replay dta.

Reply:
afaics both of these filtering test cases could be combined on replay

Reply:
yeah looking back that was a lot of noise and redundancy. did some culling and used config sources to kill noise without sacrificing coverage. thanks :+1: